activation/activation_test.go

```diff
@@ -481,414 +347,313 @@ func TestBuilder_RestartSmeshing(t *testing.T) {
-	r.NoError(err)
-	published, _, err = publishAtx(t, b, postGenesisEpoch+1, layersPerEpoch)
-	r.NoError(err)
-	r.True(published)
-	assertLastAtx(r, vPublishedAtx, vPublishedAtx, layersPerEpoch)
+	currLayer = (posAtxLayer.GetEpoch() + 1).FirstLayer()
+	atx2, err := publishAtx(t, tab, atx1.ID(), atx1.PubLayerID, &currLayer, layersPerEpoch, nil)
+	require.NoError(t, err)
+	require.NotEqual(t, atx1, atx2)
+	require.Equal(t, atx1.TargetEpoch()+1, atx2.TargetEpoch())
 }
 
 func TestBuilder_PublishActivationTx_FaultyNet(t *testing.T) {
-	r := require.New(t)
-
-	// setup
-	cdb := newCachedDB(t)
-	atxHdlr := newAtxHandler(t, cdb)
-	challenge := newChallenge(1, prevAtxID, prevAtxID, postGenesisEpochLayer, nil)
-	prevAtx := newAtx(t, challenge, sig, nipost, 2, coinbase)
-	vAtx, err := prevAtx.Verify(0, 1)
-	r.NoError(err)
-	require.NoError(t, atxHdlr.StoreAtx(context.Background(), vAtx))
-
-	cfg := Config{
-		CoinbaseAccount: coinbase,
-		GoldenATXID:     goldenATXID,
-		LayersPerEpoch:  layersPerEpoch,
-	}
-
-	ctrl := gomock.NewController(t)
-	mockSyncer := NewMocksyncer(ctrl)
-	mockSyncer.EXPECT().RegisterForATXSynced().DoAndReturn(func() chan struct{} {
-		ch := make(chan struct{})
-		close(ch)
-		return ch
-	}).AnyTimes()
-
-	// create and attempt to publish ATX
-	faultyNet := &FaultyNetMock{retErr: true}
-	b := NewBuilder(cfg, sig.NodeID(), sig, cdb, atxHdlr, faultyNet, nipostBuilderMock, &postSetupProviderMock{}, layerClockMock, mockSyncer, logtest.New(t).WithName("atxBuilder"))
-	b.commitmentAtx = &goldenATXID
-	published, _, err := publishAtx(t, b, postGenesisEpoch, layersPerEpoch)
-	r.EqualError(err, "broadcast: failed to broadcast ATX: faulty")
-	r.False(published)
-
-	// create and attempt to publish ATX
-	faultyNet.retErr = false
-	b = NewBuilder(cfg, sig.NodeID(), sig, cdb, atxHdlr, faultyNet, nipostBuilderMock, &postSetupProviderMock{}, layerClockMock, mockSyncer, logtest.New(t).WithName("atxBuilder"))
-	b.commitmentAtx = &goldenATXID
-	published, builtNipost, err := publishAtx(t, b, postGenesisEpoch, layersPerEpoch)
-	r.ErrorIs(err, ErrATXChallengeExpired)
-	r.False(published)
-	r.True(builtNipost)
+	tab := newTestBuilder(t)
+	posAtxLayer := types.NewLayerID(22)
+	currLayer := posAtxLayer
+	challenge := newChallenge(1, types.ATXID{1, 2, 3}, types.ATXID{1, 2, 3}, posAtxLayer, nil)
+	nipost := newNIPostWithChallenge(types.HexToHash32("55555"), []byte("66666"))
+	prevAtx := newAtx(t, tab.sig, tab.nodeID, challenge, nipost, 2, types.Address{})
+	vPrevAtx, err := prevAtx.Verify(0, 1)
+	require.NoError(t, err)
+	require.NoError(t, atxs.Add(tab.cdb, vPrevAtx, time.Now()))
+
+	publishErr := errors.New("blah")
+	built, err := publishAtx(t, tab, prevAtx.ID(), posAtxLayer, &currLayer, layersPerEpoch, publishErr)
+	require.ErrorIs(t, err, publishErr)
+	require.NotNil(t, built)
+
+	// now causing it to publish again, it should use the same atx
+	never := make(chan struct{}, 1)
+	done := make(chan struct{}, 1)
```

Comment:
NIT, I don't think these need a buffer
```suggestion
	never := make(chan struct{})
	done := make(chan struct{})
```

Reply:
it is a habit of mine to use buffered channel to avoid subtle blocking behavior. (it requires a receiver to be active before one can send data in, like a full channel).

Reply:
It doesn't harm. I just think that omitting size here shows the intent clearer (that this is just a signalling channel).

Reply:
that make sense. done